tf2onnx/function/gathernd.py

```diff
@@ -0,0 +1,137 @@
+    outter_shape = ctx.make_node("Slice",
+                                 [indices_shape.output[0]],
+                                 attr={"axes": [0], "ends": [-1], "starts": [0]},
+                                 dtypes=[TensorProto.INT64])
+    inner_shape = ctx.make_node("Slice",
+                                [indices_shape.output[0]],
+                                attr={"axes": [0], "ends": [sys.maxsize], "starts": [-1]},
+                                dtypes=[TensorProto.INT64])
+    outter_shape_sum = ctx.make_node("ReduceSum",
+                                     [outter_shape.output[0]],
+                                     attr={"axes": [0], "keepdims": 1},
+                                     dtypes=[TensorProto.INT64])
+    flatten_shape = ctx.make_node("Concat",
+                                  [outter_shape_sum.output[0], inner_shape.output[0]],
+                                  attr={"axis": 0},
+                                  dtypes=[TensorProto.INT64])
+    flatten_indices = ctx.make_node("Reshape", [indices, flatten_shape.output[0]])
+    nodes.extend([indices_shape, outter_shape, inner_shape, outter_shape_sum, flatten_shape, flatten_indices])
+    # outter loop for each index
+    # for (int i=0; i<outter_shape_sum; i++) inner_loop(params, flatten_indices[i])
+    cond_const = ctx.make_const(utils.make_name("cond"), np.ones((), dtype=np.bool))
+    dummy_const = ctx.make_const(utils.make_name("dummy"), np.ones((), dtype=np.int64))
+    trip_name = utils.make_name("i")
+    cond_name = utils.make_name("cond")
+    cond_out_name = utils.make_name("cond_out")
+    dummy_name = utils.make_name("dummy")
+    dummy_out_name = utils.make_name("dummy_out")
+    result_name = utils.make_name("res")
+    body_inputs = [helper.make_tensor_value_info(trip_name, onnx_pb.TensorProto.INT64, []),
+                   helper.make_tensor_value_info(cond_name, onnx_pb.TensorProto.BOOL, []),
+                   helper.make_tensor_value_info(dummy_name, node_dtype, [])]
+    body_outputs = [helper.make_tensor_value_info(cond_out_name, onnx_pb.TensorProto.BOOL, [],),
+                    helper.make_tensor_value_info(dummy_out_name, node_dtype, []),
+                    helper.make_tensor_value_info(result_name, node_dtype, [])]
+    body_nodes = []
+    index = ctx.make_node("Gather", [flatten_indices.output[0], trip_name], attr={"axis": 0})
+    index_squeeze = ctx.make_node("Squeeze", [index.output[0]], attr={"axes": [0]})
+    # inner loop to gather result
+    inner_loop_nodes, inner_loop = make_gathernd_inner_loop(ctx, params, index_squeeze, node_dtype)
+    body_nodes.extend([index.op, index_squeeze.op] + inner_loop_nodes)
+    body_nodes.append(utils.make_onnx_identity(cond_name, cond_out_name))
+    body_nodes.append(utils.make_onnx_identity(dummy_name, dummy_out_name))
+    body_nodes.append(utils.make_onnx_identity(inner_loop.output[0], result_name))
```

Comment:
so frequent similar call. shall we do this way:

body_nodes.extend([index.op, index_squeeze.op] + inner_loop_nodes,
                                 utils.make_onnx_identity(cond_name, cond_out_name),
                                  utils.make_onnx_identity(dummy_name, dummy_out_name),
                                  ...)